src/main/java/org/openrewrite/micrometer/TimerToObservation.java

```diff
@@ -0,0 +1,144 @@
+                        return super.visitCompilationUnit(cu, executionContext);
+                    }
+
+                    @Override
+                    public J.MethodInvocation visitMethodInvocation(J.MethodInvocation mi, ExecutionContext executionContext) {
+                        if (registerMatcher.matches(mi)) {
+                            Expression timerName = null;
+                            Expression registry = mi.getArguments().get(0);
+
+                            List<String> builder = new ArrayList<>();
+                            List<Expression> parameters = new ArrayList<>();
+
+                            Expression maybeBuilder = mi.getSelect();
+                            while (maybeBuilder instanceof J.MethodInvocation) {
+                                J.MethodInvocation builderMethod = (J.MethodInvocation) maybeBuilder;
+                                if (builderMatcher.matches(maybeBuilder)) {
+                                    timerName = builderMethod.getArguments().get(0);
+                                }
+                                else if (tagMatcher.matches(maybeBuilder)) {
+                                    builder.add("\n.lowCardinalityKeyValue(#{any(String)}, #{any(String)})");
+                                    parameters.add(builderMethod.getArguments().get(0));
+                                    parameters.add(builderMethod.getArguments().get(1));
+                                }
+                                else if (tagsIterableMatcher.matches(maybeBuilder)) {
+                                    builder.add("\n.lowCardinalityKeyValues(KeyValues.of(#{any(Iterable)}, Tag::getKey, Tag::getValue))");
+                                    parameters.addAll(builderMethod.getArguments());
+                                }
+                                else if (tagsMatcher.matches(maybeBuilder)) {
+                                    String args = StringUtils.repeat("#{any(String)},", builderMethod.getArguments().size());
+                                    args = args.substring(0, args.length() - 1);
+                                    builder.add("\n.lowCardinalityKeyValues(KeyValues.of(" + args + "))");
+                                    parameters.addAll(builderMethod.getArguments());
+                                }
+                                maybeBuilder = ((J.MethodInvocation) maybeBuilder).getSelect();
+                            }
+                            if (timerName != null) {
+                                parameters.add(0, timerName);
+                                parameters.add(1, registry);
+
+                                maybeAddImport("io.micrometer.observation.Observation");
+                                maybeAddImport("io.micrometer.common.KeyValues");
+                                maybeAddImport("io.micrometer.core.instrument.Tag");
+                                maybeRemoveImport("io.micrometer.core.instrument.Timer");
```

Comment:
We can move some of these the into the if conditionals above right? Such that we only evaluate them (once) if needed. Or do you prefer to keep these here?